commands/src/main/java/discord4j/commands/CommandBootstrapper.java

```diff
@@ -0,0 +1,154 @@
+package discord4j.commands;
+
+import discord4j.core.DiscordClient;
+import discord4j.core.event.domain.message.MessageCreateEvent;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+
+import java.util.function.Function;
+
+/**
+ * The (thread-safe) entry point for actually using commands. To use, simple construct a
+ * {@link discord4j.commands.CommandBootstrapper.Config} instance and call
+ * {@link discord4j.commands.CommandBootstrapper.Config#strapTo(discord4j.core.DiscordClient)}.
+ *
+ * To register events, obtain the {@link discord4j.commands.CommandDispatcher} via {@link #getCommandDispatcher()}
+ * or create a new {@link discord4j.commands.CommandDispatcher} instance replace the default instance with the new
+ * one via {@link #replaceCommandDispatcher(CommandDispatcher)}.
+ *
+ * @see #detach()
+ * @see discord4j.commands.CommandBootstrapper.Config
+ * @see discord4j.commands.CommandProvider
+ * @see discord4j.commands.CommandDispatcher
+ * @see discord4j.commands.CommandErrorHandler
+ * @see discord4j.commands.DefaultCommandDispatcher
+ */
+public final class CommandBootstrapper {
```

Comment:
I'm not sure, we've gone back and forth quite a bit. Mutability in this class isn't horrible since it essentially only contains orchestration logic and its functionality is a composition of its components. So it would be easier for a user to modify this class rather than rebuild it--particularly in the case of an ecosystem where users are given a pre-built instance via dependency injection or something.

Reply:
That reconstruction could also be handled by that "ecosystem" but I'm fine with this for now.